modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/jms/Acknowledge.java

```diff
@@ -48,14 +51,33 @@
 @BallerinaAnnotation(annotationName = "Param", attributes = { @Attribute(name = "deliveryStatus",
         value = "Specify whether message delivery is SUCCESS or ERROR") })
 public class Acknowledge extends AbstractNativeFunction {
+    private static final Logger log = LoggerFactory.getLogger(Acknowledge.class);
+
     public BValue[] execute(Context ctx) {
         BMessage msg = (BMessage) getArgument(ctx, 0);
         CarbonMessage carbonMessage = msg.value();
         String deliveryStatus = getArgument(ctx, 1).stringValue();
+        Object jmsSessionAcknowledgementMode = carbonMessage
+                .getProperty(JMSConstants.JMS_SESSION_ACKNOWLEDGEMENT_MODE);
 
-        if (ctx.getBalCallback() != null) {
-            carbonMessage.setProperty(JMSConstants.JMS_MESSAGE_DELIVERY_STATUS, deliveryStatus);
-            ctx.getBalCallback().done(carbonMessage);
+        if (null == jmsSessionAcknowledgementMode) {
+            log.warn("JMS Acknowledge function can only be used with JMS Messages. "
```

Comment:
Do we have to do a "WARN" here? If the property is optional, then we can check and act upon it, rather than do a "WARN"